### PR TITLE
Make juju-gui-charm a fat charm by including the release and jujugui dependencies

### PR DESCRIPTION
**Pull request by bac**
The requirement that we deploy in production from Launchpad means we must support a fat charm in version control, abandoning our attempt to simply package the charm and upload directly to github.

### CODE REVIEW

**Comment by bac** [comment]
This branch updates the makefile to not assume `releases` and `jujugui-deps` are non-existent but allows the contents to be updated in place by the `make package` command.

Handling of packaging for precise is a special case as we never want to check in the dependencies it generates as they are different than for the other charms.  We will continue to create an temporary set of dependencies and then upload directly to the charmstore and then discard the changes.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/30/
Test PASSed.

**Comment by mhilton** [comment]
:+1:

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -112,6 +115,15 @@ required testing, and then publish it to make the release:
 
     charm publish cs:development/juju-gui
 
+## Releasing to Launchpad for production ##
+
+When the charm is ready to be released to production, it must be packaged,
+commited to github, QA'd, reviewed, and merged.  It must also be pushed to
+Launchpad using the following:
+
+    git remote add lporigin lp:~yellow/canonical-theblues-charms/+git/juju-gui
```
Comment (frankban) [inline RELEASE_PROCESS.md]: Nice.

**Comment by frankban** [comment]
:+1: thanks!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/32/
Test PASSed.

**Comment by bac** [comment]
Thanks for the reviews @mhilton and @frankban 

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-charm-merge

**Pull request merged**